### PR TITLE
fix(errors): add valid process exit value on error #130

### PR DESCRIPTION
**Pull request by codeflyer**
In this PR:

When some CDK action fails the mira command returns 1 as exit code. This allow the CICD to stop correctly in case of error.

The error management in the bootstrap method was normalised.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -2,4 +2,14 @@
 require('source-map-support').install()
 
 const { MiraBootstrap } = require('../dist/src/cdk/bootstrap')
-new MiraBootstrap().initialize()
+
+async function run () {
+  try {
+    await new MiraBootstrap().initialize()
+  } catch (e) {
+    console.error(e)
```
Comment (mikegwhit) [inline bin/cli.js:10]: Hm, thinking that there should for sure be other try / catch blocks within `bootstrap.ts` an `app.ts`.  See `app.ts` around line # 164

File: src/cdk/bootstrap.ts [hunk 2]
```diff
@@ -212,40 +210,50 @@ export class MiraBootstrap {
         ...process.env
       }
     })
-    await new Promise(resolve => {
-      proc.on('exit', () => {
+    await new Promise((resolve, reject) => {
```
Comment (mikegwhit) [inline src/cdk/bootstrap.ts:213]: Wondering if this `deployCi` fn is where we want to put the try/catch block instead (see above remark)?

**Review by mikegwhit — APPROVED** [review]
I see how we're now bubbling the errors up to the invoking CLI proc in this PR.  Just seeking clarification on what wasn't working before with it or am wondering if there's additional error handling we want to do within the application.  Otherwise LGTM.

File: src/cdk/bootstrap.ts (hunk 2 above)
Comment (mikegwhit) [inline src/cdk/bootstrap.ts:213]: Is it possible here to catch the rejection (if it occurs) and provide a meaningful message before exiting (presumably with the more detailed error message)?

i.e. "Deploy CICD Failed.  Dumping Error message"

and then the CLI try/catch block presumably will exit with a non-zero code and dump the error details.

**Review by mikegwhit — DISMISSED** [review]

File: src/cdk/bootstrap.ts (hunk 2 above)
Comment (codeflyer) [reply src/cdk/bootstrap.ts:215]: I've just added an error message to each failing point to return the place where the error actually happened. 

Collecting the errors and print the dump is a bit difficult here, it's child process that has already printed the errors before exit.

If we would like to have a better error management we should probably consider a specific task that goes through the whole process find a proper way to collect the errors.

Comment (mikegwhit) [reply src/cdk/bootstrap.ts:215]: Sounds good to me.  LGTM

**Review by mikegwhit — APPROVED** [review]

**Pull request merged**